pymysqlreplication/tests/test_basic.py

```diff
@@ -1473,6 +1472,39 @@ def test_rows_query_log_event(self):
         self.assertIsInstance(event, RowsQueryLogEvent)
 
 
+class TestGtidEvent(base.PyMySQLReplicationTestCase):
+    def setUp(self):
+        super(TestGtidEvent, self).setUp()
+        self.execute("SET SESSION binlog_rows_query_log_events=1")
+
+    def tearDown(self):
+        self.execute("SET SESSION binlog_rows_query_log_events=0")
+        super(TestGtidEvent, self).tearDown()
+
+    def test_gtid_event(self):
+        self.stream.close()
+        self.stream = BinLogStreamReader(
+            self.database,
+            server_id=1024,
+            only_events=[GtidEvent],
+        )
+        self.execute(
+            "CREATE TABLE IF NOT EXISTS test (id INT AUTO_INCREMENT PRIMARY KEY, name VARCHAR(255))"
+        )
+        gtid_event = self.stream.fetchone()
+
+        self.assertIsInstance(gtid_event, GtidEvent)
+        self.assertIsInstance(gtid_event.event_type, int)
+        self.assertIsInstance(gtid_event.sid, bytes)
+        self.assertIsInstance(gtid_event.gno, int)
+        self.assertIsInstance(gtid_event.lt_type, int)
+        self.assertIsInstance(gtid_event.last_committed, int)
```

Comment:
Upper mysql 5.7 version has this variable so pytest failed.
1.  base.py make method **isMySQL57AndMore**
2.  setUp method  use *isMySQL57AndMore** before start Test

Reply:
Resolves it!